plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -0,0 +1,245 @@
+                .map(ignored -> singletonList(prefix.toSchemaTableName()))
+                .orElseGet(() -> listTables(session, prefix.getSchema()));
+        return tables.stream()
+                .map(table -> getTableColumnMetadata(session, table))
+                .flatMap(Optional::stream)
+                .iterator();
+    }
+
+    HiveMetastore getMetastore()
+    {
+        return metastore;
+    }
+
+    private boolean isHudiTable(ConnectorSession session, Table table)
+    {
+        String basePath = table.getStorage().getLocation();
+        Configuration configuration = hdfsEnvironment.getConfiguration(new HdfsContext(session), new Path(basePath));
+        try {
+            checkTableValidity(getFs(basePath, configuration), new Path(basePath), new Path(basePath, METAFOLDER_NAME));
+        }
+        catch (org.apache.hudi.exception.TableNotFoundException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private Optional<TableColumnsMetadata> getTableColumnMetadata(ConnectorSession session, SchemaTableName table)
+    {
+        try {
+            List<ColumnMetadata> columns = getTableMetadata(table, getColumnsToHide(session)).getColumns();
+            return Optional.of(TableColumnsMetadata.forTable(table, columns));
+        }
+        catch (TableNotFoundException ignored) {
+            return Optional.empty();
+        }
+    }
+
+    private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName, Collection<String> columnsToHide)
+    {
+        Table table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(tableName));
+        Function<HiveColumnHandle, ColumnMetadata> metadataGetter = columnMetadataGetter(table);
+        List<ColumnMetadata> columns = hiveColumnHandles(table, typeManager, NANOSECONDS).stream()
+                .filter(column -> !columnsToHide.contains(column.getName()))
+                .map(metadataGetter)
+                .collect(toImmutableList());
+
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+
+        // Partitioning property
+        List<String> partitionedBy = table.getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+        if (!partitionedBy.isEmpty()) {
+            properties.put(PARTITIONED_BY_PROPERTY, partitionedBy);
+        }
+
+        Optional<String> comment = Optional.ofNullable(table.getParameters().get(TABLE_COMMENT));
+        return new ConnectorTableMetadata(tableName, columns, properties.buildOrThrow(), comment);
+    }
+
+    private List<String> listSchemas(ConnectorSession session, Optional<String> schemaName)
+    {
+        return schemaName
+                .filter(name -> !isHiveSystemSchema(name))
+                .map(ImmutableList::of)
+                .orElseGet(() -> (ImmutableList<String>) listSchemaNames(session));
```

Comment:
This cast is causing issues when showing the tables from `information_schema`: 

```
trino:default> show tables in hudi.information_schema;
Query 20220719_045752_00010_iyr2a failed: class java.util.ArrayList cannot be cast to class com.google.common.collect.ImmutableList (java.util.ArrayList is in module java.base of loader 'bootstrap'; com.google.common.collect.ImmutableList is in unnamed module of loader io.trino.server.PluginClassLoader @642334d6)
java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class com.google.common.collect.ImmutableList (java.util.ArrayList is in module java.base of loader 'bootstrap'; com.google.common.collect.ImmutableList is in unnamed module of loader io.trino.server.PluginClassLoader @642334d6)
	at io.trino.plugin.hudi.HudiMetadata.lambda$listSchemas$7(HudiMetadata.java:243)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at io.trino.plugin.hudi.HudiMetadata.listSchemas(HudiMetadata.java:243)
	at io.trino.plugin.hudi.HudiMetadata.listTables(HudiMetadata.java:165)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.listTables(ClassLoaderSafeConnectorMetadata.java:286)
	at io.trino.metadata.MetadataManager.listTables(MetadataManager.java:496)
	at io.trino.metadata.MetadataListing.listTables(MetadataListing.java:101)
```

Reply:
will fix it. i think it was suggested in one of the earlier comments.